spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindingService.java

```diff
@@ -118,13 +120,35 @@ public BindingService(BindingServiceProperties bindingServiceProperties,
 			String[] bindingTargets = StringUtils
 					.commaDelimitedListToStringArray(bindingTarget);
 			for (String target : bindingTargets) {
-				Binding<T> binding = input instanceof PollableSource
+				if (ObjectUtils.isEmpty(consumerProperties.getInstanceIndexList())) {
```

Comment:
I believe we should only do this for partitioned consumers.

				if (!consumerProperties.isPartitioned() || ObjectUtils.isEmpty(consumerProperties.getInstanceIndexList())) {